engine/access/apiproxy/access_api_proxy.go

```diff
@@ -0,0 +1,519 @@
+package apiproxy
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc/connectivity"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/utils/grpcutils"
+)
+
+// FlowAccessAPIRouter is a structure that represents the routing proxy algorithm.
+// It splits requests between a local and a remote API service.
+type FlowAccessAPIRouter struct {
+	Logger   zerolog.Logger
+	Metrics  *metrics.ObserverCollector
+	Upstream *FlowAccessAPIForwarder
+
+	// local observer
+	access.AccessAPIServer
+}
+
+func (h *FlowAccessAPIRouter) log(rpc, handler string, err error) {
```

Comment:
I think these are reversed

```suggestion
func (h *FlowAccessAPIRouter) log(handler, rpc string, err error) {
```